src/gaxios.ts

```diff
@@ -219,21 +237,25 @@ export class Gaxios {
     if (opts.data) {
       if (isStream.readable(opts.data)) {
         opts.body = opts.data;
+      } else if (hasBuffer() && Buffer.isBuffer(opts.data)) {
+        // Do not attempt to JSON.stringify() a Buffer:
+        opts.body = opts.data;
+        if (!hasHeader(opts, 'Content-Type')) {
+          opts.headers['Content-Type'] = 'application/json';
```

Comment:
this also should be `getHeader` to ensure case safety.  I am starting to wonder if we should lowercase all header names in `validateOptions`, and then _always_ use lowercase throughout this code.  Eh?

Reply:
This doesn't need to be, because we already know it doesn't have the header `Content-Type`, or `content-type`. So, we're just picking one when we choose it.

Reply:
oh, for clarification `hasHeader` doesn't care about case.